Week 1 663380145-7 นางสาวกุลศยา จันภูงา

Lab 1.1: Environment Setup & First Script


In [30]:
print("Hello world")
print("My Name Is Kunsaya Junpoong ")
print("Student ID : 663380145-7")

Hello world
My Name Is Kunsaya Junpoong 
Student ID : 663380145-7


Lab 1.2: Working with Variables and Basic Data Types
- Objective: Students will define variables of different data types, perform simple operations, and display results.


In [31]:
Age = 21
GPA = 4.00
Favorite_Color = "Red"
Student = True

print("Age = ",Age )
print("GPA = ",GPA)
print("Favorite_Color = ",Favorite_Color)
print("Student = ",Student)

Age =  21
GPA =  4.0
Favorite_Color =  Red
Student =  True


3. Perform simple arithmetic:

In [32]:
num1 = 8
num2 = 9
sum = (num1 + num2) / (num2 - num1) * num1
print("Sum = ",sum)

Sum =  136.0


4. String concatenation and f-strings:

In [33]:
First_name = "Kunsaya"
Last_name = "Junpoonga"
Full_name = First_name + " " + Last_name
print(f"Miss {Full_name}")

Miss Kunsaya Junpoonga


5. User input and type conversion:


In [34]:
num = input("What your favorite number : ")
print("type of A is : ", type(num))
num = int(num)
result = num + 5
print(num," + 5 = ", result)

What your favorite number : 19
type of A is :  <class 'str'>
19  + 5 =  24


Activity 1.1: Class Discussion - Uses of Scripting

การเขียนสคริปต์ด้วยภาษา Python สามารถนำไปประยุกต์ใช้ในสถานการจริงได้ ดังนี้

1.   พัฒนาเว็บไซต์ สร้างเว็บไซต์และระบบหลังบ้าน
2.   พัฒนาเกมและแอปพลเคชัน สร้างเกมขนาดเล็ก พัฒนาโปรแกรมสำหรับใช้งานบนคอมพิวเตอร์
3.   วิเคราะห์ข้อมูล คำนวณ สรุปและสร้างกราฟข้อมูลจำนวนมาก
4.   การดึงข้อมูลจากเว็บ (Web Scraping): สามารถเขียนสคริปต์เพื่อดึงข้อมูลจากหน้าเว็บไซต์ต่างๆ มาใช้งาน
5.   การสื่อสารอัตโนมัติ (Automated Communications): สามารถใช้ Python ในการส่ง Email และ SMS แบบอัตโนมัติ

Activity 1.2: "Python Puzzlers" - Identifying Data Types & Output

In [35]:
# Puzzler 1
x = 10
y = 3.5
z = x + y
print(z)
print(type(z))

13.5
<class 'float'>


**Puzzler 1**

ได้ผลลัพธ์: 13.5

DataType : float

ทำไมถึงเป็นtype float เนื่องจาก เมื่อนำint + float ภาษาPython จะแปลงเป็น floatโดยอัตโนมัติ

In [36]:
# Puzzler 2
name = "Alice"
age = "30" # Note: This is a string
message = name + " is " + age + " years old."
print(message)
print(type(age))

Alice is 30 years old.
<class 'str'>


**Puzzler 2**

ได้ผลลัพธ์: 30

DataType : string

ทำไมถึงเป็นtype string เนื่องจาก ตอนที่ประกาศตัวแปร age = "30" ใช้ Double Quotes จึงถือว่าเป็น String

In [37]:
# Puzzler 3
is_sunny = True
is_raining = False
print(is_sunny and is_raining)
print(type(is_sunny))

False
<class 'bool'>


**Puzzler 3**

ได้ผลลัพธ์: false

DataType : boolean

ทำไมถึงเป็นtype boolean เนื่องจาก ตอนประกาศตัวแปร True,False ไม่ได้ใส่ Double Quotes,Single Quotes ดังนั้นจึงนับเป็น Boolean โดยอัตโนมัติ

In [38]:
# Puzzler 4
# What happens here?
# num_str = input("Enter a number: ")
# result = num_str + 5
# print(result)

**Puzzler 4**

ได้ผลลัพธ์: TypeError

เนื่องจาก ฟังก์ชัน input จะรับค่าเป็น String เสมอ และไม่สามารถนำ String + Int ได้ กังนั้นจึงError หากต้องการรับInput เป็น Number จะต้องแปลงจากString เป็นInt or Floatก่อน